src/include/net_message.h

```diff
@@ -95,6 +96,17 @@ class CServerSetup
 	uint8_t Ready[PlayerMax];      /// Client ready state
 	uint8_t Race[PlayerMax];       /// Client race selection
 	// Fill in here...
+
+	union {
+		uint8_t BitField; // IMPORTANT! must be the same size as the struct below
+		struct Fields {
+			uint8_t AiChecksDependencies : 1;
+			uint8_t AiExplores : 1;
+			uint8_t AllyDepositsAllowed : 1;
+			uint8_t SimplifiedAutoTargeting : 1;
+		} Fields;
+		static_assert(sizeof(BitField) == sizeof(struct Fields), "bitfield and storage of shared prefs");
+	} SharedPreferences;
```

Comment:
Don't you think it's very hardcoded for war1gus and wargus? I mean - stratagus - it's an engine. Maybe to do it in some more general way? 
For example declare in the lua config files which preferences could cause the desync and sync them before game starts.

Reply:
For example "allow multiple town halls" is actual only for war1gus, but has to be synced here too.